packages/react-dom/src/__tests__/ReactDOMNativeEventHeuristic-test.js

```diff
@@ -383,17 +383,13 @@ describe('ReactDOMNativeEventHeuristic-test', () => {
     });
     expect(container.textContent).toEqual('Count: 0');
 
-    const pressEvent = document.createEvent('Event');
-    pressEvent.initEvent('click', true, true);
-    dispatchAndSetCurrentEvent(target, pressEvent);
-
-    assertLog(['Count: 0 [after batchedUpdates]']);
-    expect(container.textContent).toEqual('Count: 0');
-
-    // Intentionally not using `act` so we can observe in between the click
```

Comment:
Ditto here

Reply:
The comment is a bit sloppily worded. The reason to use `await null` instead of `act` in these tests is to test that the click is processed synchronously in a microtask, as opposed to a Scheduler task / `postTask`. But this test is not really about that, it's about whether the updates were batched. In this test it's sufficient to check before the end of the `act` scope, which is akin to checking at the end of an event handler. (Which would have been the more realistic way to write this test in the first place.)